adapters/koa/koa.ts

```diff
@@ -29,18 +32,31 @@ function adapter(router: Router): runtime.server.ServerAdapter {
         servers: [],
         op,
         headers: ctx.request.headers,
-        params: ctx.params,
+        params: (ctx as any).params,
         query: ctx.query,
-        body
+        body,
+        requestContext: requestContextCreator(ctx)
       });
       ctx.status = result.status;
       ctx.body = result.value.value;
     });
   };
 }
 
-export function bind<Spec>(handler: runtime.server.HandlerFactory<Spec>, spec: Spec): Router {
-  const router = new Router();
-  handler(adapter(router))(spec);
+/**
+ * Bind provided handlers for the OpenAPI routes
+ *
+ * Koa's default StateT and CustomT values are selected as defaults
+ * @param handler
+ * @param spec
+ * @param requestContextCreator
+ */
+export function bind<Spec, RequestContext = any, StateT = any, CustomT = {}>(
```

Comment:
could these anys be unknowns or voids by default?

Reply:
RequestContext I could probably switch to void, but StateT, and CustomT types are what they are right now by the default in Koa. We can change them to voids, but then default return type will change from what it is right now.